catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -103,14 +110,17 @@
    *     if Apache changes this ModelCamelContext interface there is no guarantee that whatever DM
    *     is being used (Blueprint in this case) will be updated accordingly.
    */
-  public ContentDirectoryMonitor(CamelContext camelContext, AttributeRegistry attributeRegistry) {
+  public ContentDirectoryMonitor(
```

Comment:
✏️ Is the overloaded constructors needed for `ContentDirectoryMonitor` and `CdmServiceWaiter`? Can the blueprint and constructors be condensed to something like

```java
public ContentDirectoryMonitor(
    CamelContext camelContext,
    AttributeRegistry attributeRegistry,
    int maxRetries,
    int delayBetweenRetries,
    Executor configurationExecutor,
    CdmServiceWaiter cdmServiceWaiter,
    CdmState initState) {
  this.camelContext = camelContext;
  this.attributeRegistry = attributeRegistry;
  this.maxRetries = maxRetries;
  this.delayBetweenRetries = delayBetweenRetries;
  this.configurationExecutor = configurationExecutor;
  this.cdmServiceWaiter = cdmServiceWaiter;
  this.initState = initState;
  setBlacklist();
}
```
```java
public CdmServiceWaiter(
    String propertyId,
    String className,
    int timeout,
    TimeUnit timeUnit,
    ScheduledExecutorService scheduledExecutorService,
    OsgiServices osgiServices,
    CdmState cdmState) {
  this.propertyId = propertyId;
  this.timeout = timeout;
  this.timeUnit = timeUnit;
  this.className = className;
  this.scheduledExecutorService = scheduledExecutorService;
  this.osgiServices = osgiServices;
  this.initState = cdmState;
}
```
```xml
<bean id="cdmState" class="org.codice.ddf.catalog.content.monitor.CdmState"/>
<cm:managed-service-factory
    id="contentDirectoryMonitor_MSF"
    factory-pid="org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor"
    interface="org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor">
    <cm:managed-component class="org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor"
                          init-method="init" destroy-method="destroy">
        <argument ref="camelContext"/><!--CamelContext camelContext-->
        <argument ref="attributeRegistry"/><!--AttributeRegistry attributeRegistry-->
        <argument value="20"/><!--int maxRetries-->
        <argument value="5"/><!--int delayBetweenRetries-->
        <argument><!--Executor configurationExecutor-->
            <bean class="java.util.concurrent.Executors"
                  factory-method="newSingleThreadScheduledExecutor">
                <argument>
                    <bean class="org.codice.ddf.platform.util.StandardThreadFactoryBuilder"
                          factory-method="newThreadFactory">
                        <argument value="contentDirectoryMonitorThread"/>
                    </bean>
                </argument>
            </bean>
        </argument>
        <argument><!--CdmServiceWaiter cdmServiceWaiter-->
            <bean class="org.codice.ddf.catalog.content.monitor.osgi.CdmServiceWaiter">
                <argument value="id"/><!--String propertyId-->
                <argument value="InputTransformer"/><!--String className-->
                <argument value="1"/><!--int timeout-->
                <argument><!--TimeUnit timeUnit-->
                    <bean class="java.util.concurrent.TimeUnit" factory-method="valueOf">
                        <argument value="MINUTES"/>
                    </bean>
                </argument>
                <argument><!--ScheduledExecutorService scheduledExecutorService-->
                    <bean class="java.util.concurrent.Executors"
                          factory-method="newSingleThreadScheduledExecutor">
                        <argument>
                            <bean class="org.codice.ddf.platform.util.StandardThreadFactoryBuilder"
                                  factory-method="newThreadFactory">
                                <argument value="CdmServiceWaiterThread"/>
                            </bean>
                        </argument>
                    </bean>
                </argument>
                <argument><!--OsgiServices osgiServices-->
                    <bean class="org.codice.ddf.catalog.content.monitor.osgi.OsgiServices"/>
                </argument>
                <argument ref="cdmState"/><!--CdmState cdmState-->
            </bean>
        </argument>
        <argument ref="cdmState"/><!--CdmState initState-->
        <property name="numThreads" value="1"/>
        <property name="readLockIntervalMilliseconds" value="500"/>
        <property name="monitoredDirectoryPath" value=""/>
        <property name="attributeOverrides">
            <list/>
        </property>
        <cm:managed-properties persistent-id=""
                               update-strategy="component-managed"
                               update-method="updateCallback"/>
    </cm:managed-component>
</cm:managed-service-factory>
```
✏️ Could the `ContentDirectoryMonitor` get the initState from the `CdmServiceWaiter` instead of getting it from a constructor argument?

Reply:
I need to pass the same and a new CdmState into each the CDM and CdmServiceWaiter. Not sure if this will allow me to do it.

Reply:
The xml that I suggested does pass the same new CdmState into each the CDM and CdmServiceWaiter.